server/src/main/scala/org/http4s/server/middleware/CSRF.scala

```diff
@@ -406,10 +407,26 @@ object CSRF {
       }
     }
 
+    def getMultipartToken: F[Option[String]] =
+      if (r.headers.get(headers.`Content-Type`).map(_.mediaType.isMultipart).getOrElse(false)) {
+        val token = multipart[G].decode(r, strict = true).value.flatMap {
+          case Right(dec) =>
+            // The `.drop(1)` removes the first empty line from the body value of the field.
+            dec.parts
+              .find(_.headers.exists(_.value.contains(s"""name="$fieldName"""")))
```

Comment:
I think we should be looking for Content-Disposition headers here?  Also, without double checking the RFC, I'm not certain the name needs to be quoted.  The parser should handle that.

Reply:
Right. Regarding the quotes I found nothing specific in RFC-7578, but their example uses quotes: `Content-Disposition: form-data; name="user"`
It also mentions that the field name *could* be encoded but I guess that would have already handled here, or is it?

Reply:
Not compiled, but I think this does what you want:

```
dec.parts.find(_.headers.get(`Content-Disposition`).fold(false, _.get("name").isDefined)))
```